File: leet-code/242. Valid Anagram/ValidAnagram242.java
package com.example;

import java.util.stream.Collectors;

public class ValidAnagram242 {
    public boolean isAnagram(String s, String t) {
        return isSame(s, t);
    }

    private boolean isSame(String s, String t) {
        // https://www.baeldung.com/java-string-to-stream
        String sortedS = s
                .chars()
                .sorted()
                .mapToObj(c -> (char) c)
                .map(String::valueOf)
                .collect(Collectors.joining());

        String sortedT = t
                .chars()
                .sorted()
                .mapToObj(c -> (char) c)
                .map(String::valueOf)
                .collect(Collectors.joining());

        return sortedS.equals(sortedT);
    }
}

File: leet-code/242. Valid Anagram/ValidAnagram242.kt
package com.example

class ValidAnagram242 {
    fun isAnagram(s: String, t: String): Boolean {
        // counting 정렬 활용
        val sCount = IntArray(26)
        val tCount = IntArray(26)
        for (sChar in s) {
            sCount[sChar - 'a'] += 1
        }
        for (tChar in t) {
            tCount[tChar - 'a'] += 1
        }
        for (i in 0..25) {
            if (sCount[i] != tCount[i]) {
                return false
            }
        }
        return true
    }
}